File: AppExamples/Java/app/src/main/java/com/android/starter/network/Repository.java
package com.android.starter.network;

import com.android.starter.BuildConfig;

import java.io.IOException;
import retrofit2.Retrofit;
import retrofit2.converter.scalars.ScalarsConverterFactory;

public class Repository {

    private final String ip = "192.168.184.195";
    private final String baseUrl = "http://$ip:3000/";

    private final String FRAMEWORK = "JAVA";
    private final String TEST_TYPE = "example";
    private final String ACTIVITY = ".MainActivity";

    Api service = new Retrofit.Builder()
            .baseUrl(baseUrl)
            .addConverterFactory(ScalarsConverterFactory.create())
            .build()
            .create(Api.class);

    public String started() throws IOException {
        return service.started(
                android.os.Build.MODEL,
                TEST_TYPE,
                BuildConfig.APPLICATION_ID,
                ACTIVITY,
                FRAMEWORK
        ).execute().body();
    }

    public String logData() throws IOException {
        return service.logData(
                android.os.Build.MODEL,
                TEST_TYPE,
                BuildConfig.APPLICATION_ID,
                ACTIVITY,
                FRAMEWORK
        ).execute().body();
    }

    public String done() throws IOException {
        return service.done(
                android.os.Build.MODEL,
                TEST_TYPE,
                BuildConfig.APPLICATION_ID,
                ACTIVITY,
                FRAMEWORK
        ).execute().body();
    }

}

File: AppExamples/Kotlin/app/src/main/java/com/android/starter/network/Repository.kt
package com.android.starter.network

import android.util.Log
import com.android.starter.BuildConfig
import retrofit2.Retrofit
import retrofit2.converter.scalars.ScalarsConverterFactory

class Repository {

    private val ip = "192.168.1.68"
    private val baseUrl = "http://$ip:3000/"

    private val service = Retrofit.Builder()
        .baseUrl(baseUrl)
        .addConverterFactory(ScalarsConverterFactory.create())
        .build()
        .create(Api::class.java)

    suspend fun started(): String {
        Log.i("REQUEST", "WHATNOW")
        val call = service.started(
            device = android.os.Build.MODEL,
            appId = BuildConfig.APPLICATION_ID
        )
        return call.body() ?: throw ApiError(call.errorBody()!!.toString())
    }

    suspend fun logData(): String {
        Log.i("REQUEST", "LOGDATA")
        val call = service.logData(
            device = android.os.Build.MODEL,
            appId = BuildConfig.APPLICATION_ID
        )
        return call.body() ?: throw ApiError(call.errorBody()!!.toString())
    }

    suspend fun done(): String {
        Log.i("REQUEST", "DONE")
        val call = service.done(
            device = android.os.Build.MODEL,
            appId = BuildConfig.APPLICATION_ID
        )
        return call.body() ?: throw ApiError(call.errorBody()!!.toString())
    }

    suspend fun test(): String {
        val call = service.test()
        return call.body() ?: throw ApiError(call.errorBody()!!.toString())
    }
}

class ApiError(body: String): Exception(body)